bundle/jsky.app.ot/src/test/scala/jsky/app/ot/gemini/editor/sitequality/TimingWindowImporterSpec.scala

```diff
@@ -0,0 +1,110 @@
+package jsky.app.ot.gemini.editor.sitequality
+
+import java.time.Instant
+
+import edu.gemini.spModel.gemini.obscomp.SPSiteQuality.TimingWindow
+import jsky.app.ot.gemini.editor.sitequality.TimingWindowParser.{TimingWindowParseFailure, TimingWindowParseResults}
+import org.specs2.ScalaCheck
+import org.specs2.mutable.Specification
+
+import scalaz._
+import Scalaz._
+
+object TimingWindowImporterSpec extends Specification with ScalaCheck {
+  "timing window parsing" should {
+    "handle empty inputs" in {
+      val results = TimingWindowParser.parseTimingWindows(Nil)
+      (results.successes.isEmpty must beTrue) and (results.failures.isEmpty must beTrue)
+    }
+
+    "ignore empty lines and comments" in {
+      val lines = List("", "  ", "#", "# Comment", " # abc ", "    ## nothing here")
+      val results = TimingWindowParser.parseTimingWindows(lines)
+      (results.successes.isEmpty must beTrue) and (results.failures.isEmpty must beTrue)
+    }
+
+    "create timing windows from complete data, ignoring whitespace" in {
+      val t1Tup = ("2018-01-01", "01:02:03", "04:05", 1.some, "01:02:03".some)
```

Comment:
Due to repetitions, I think, dates in the past should be acceptable, as even if the date is in the past, with repetitions, the periods defined by the info could extend (possibly infinitely) into the future.